options/m_option.c

```diff
@@ -2381,6 +2381,58 @@ const m_option_type_t m_option_type_size_box = {
     .equal = geometry_equal,
 };
 
+static int parse_rect(struct mp_log *log, const m_option_t *opt,
+                      struct bstr name, struct bstr param, void *dst)
+{
+    bool is_help = bstr_equals0(param, "help");
+    if (is_help)
+        goto exit;
+
+    int w, h, x = 0, y = 0;
+    if (bstr_sscanf(param, "%d:%d:%d:%d", &w, &h, &x, &y) < 2)
+        goto exit;
+
+    struct mp_rect rc = {
+        .x0 = x,
+        .x1 = x + w,
+        .y0 = y,
+        .y1 = y + h,
+    };
+
+    if (dst)
+        *((struct mp_rect *) dst) = rc;
+
+    return 1;
+
+exit:
+    if (!is_help) {
+        mp_err(log, "Option %.*s: invalid rect: '%.*s'\n", BSTR_P(name),
+               BSTR_P(param));
+    }
+    mp_info(log, "Valid format: [W[%%]:H[%%][:X[%%]][:Y[%%]]]\n");
+    return is_help ? M_OPT_EXIT : M_OPT_INVALID;
+}
+
+static char *print_rect(const m_option_t *opt, const void *val)
+{
+    const struct mp_rect *rc = val;
+    return talloc_asprintf(NULL, "%d:%d:%d:%d", mp_rect_w(*rc), mp_rect_h(*rc),
+                           rc->x0, rc->y0);
+}
+
+static bool rect_equal(const m_option_t *opt, void *a, void *b)
+{
+    return mp_rect_equals(a, b);
+}
+
+const m_option_type_t m_option_type_rect = {
+    .name  = "Rect",
+    .size  = sizeof(struct mp_rect),
```

Comment:
Couldn't this be based around `struct m_geometry` instead?

Reply:
I see your point, but it feels little bit off to tide that into `struct m_geometry`

> Like how the size_box type currently piggybacks off of it.

Yes, but still `parse_size_box` exists, still would need to add `parse_rect` because I want w, h be required and x, y be optional, so opposite of `--geomentry`. And parsing/using `struct mp_rect` didn't turn out to be a lot of code. If anything we would only change the type of struct, rest mostly stays. So I don't see a gain to piggyback of `m_geometry`

But if you see a good reason to do so, please tell, I might be missing something.

Reply:
Nah that's a fair point. I was just wondering.